test/functional/terminal/highlight_spec.lua

```diff
@@ -178,11 +179,20 @@ describe('synIDattr()', function()
     eq('79',  eval('synIDattr(hlID("Keyword"), "fg#")'))
     eq('',    eval('synIDattr(hlID("Keyword"), "sp")'))
     eq('',    eval('synIDattr(hlID("Keyword"), "sp#")'))
+    eq('-1',  eval('synIDattr(hlID("Visual"),  "fg")'))
+    eq('-1',  eval('synIDattr(hlID("Visual"),  "fg#")'))
+    eq('252', eval('synIDattr(hlID("Visual"),  "bg")'))
+    eq('252', eval('synIDattr(hlID("Visual"),  "bg#")'))
+    eq('',    eval('synIDattr(hlID("Visual"),  "sp")'))
+    eq('',    eval('synIDattr(hlID("Visual"),  "sp#")'))
   end)
 
   it('returns gui-color if "gui" arg is passed', function()
     eq('Black',  eval('synIDattr(hlID("Normal"),  "bg", "gui")'))
     eq('Maroon', eval('synIDattr(hlID("Keyword"), "sp", "gui")'))
+    eq('bg',     eval('synIDattr(hlID("Visual"),  "fg", "gui")'))
+    eq('fg',     eval('synIDattr(hlID("Visual"),  "bg", "gui")'))
+    eq('bg',     eval('synIDattr(hlID("Visual"),  "sp", "gui")'))
```

Comment:
Let's move these into separate `it()` blocks like this, for clarity:

```lua
it('returns "Normal" values for group using special fg/bg names', function()
  execute('highlight Visual ctermfg=bg ctermbg=fg guifg=bg guibg=fg guisp=bg')
  eq(eval('synIDattr(hlID("Normal"),  "fg", "gui")'), eval('synIDattr(hlID("Visual"),  "fg", "gui")'))
  eq(eval('synIDattr(hlID("Normal"),  "bg", "gui")'), eval('synIDattr(hlID("Visual"),  "bg", "gui")'))
  ...
end)
```

All the tests you added can live in this single test.